examples/move/big_vector/sources/big_vector.move

```diff
@@ -0,0 +1,1274 @@
+/// BigVector is an arbitrary sized vector-like data structure,
+/// implemented using an on-chain B+ Tree to support almost constant
+/// time (log base max_fan_out) random access, insertion and removal.
+///
+/// Iteration is supported by exposing access to leaf nodes (slices).
+/// Finding the initial slice can be done in almost constant time, and
+/// subsequently finding the previous or next slice can also be done
+/// in constant time.
+///
+/// Nodes in the B+ Tree are stored as individual dynamic fields
+/// hanging off the `BigVector`.
+///
+/// Note: The index type is `u128`, but the length is stored as `u64`
+/// because the expectation is that indices are sparsely distributed.
+module big_vector::big_vector {
+    use sui::dynamic_field as df;
+    use sui::object::UID;
+    use sui::tx_context::TxContext;
+
+    use fun sui::object::new as TxContext.new;
+    use fun sui::object::delete as UID.delete;
+
+    use fun big_vector::utils::pop_until as vector.pop_until;
+    use fun big_vector::utils::pop_n as vector.pop_n;
+
+    public struct BigVector<phantom E: store> has key, store {
+        id: UID,
+
+        /// How deep the tree structure is.
+        depth: u8,
+
+        /// Total number of elements that this vector contains, not
+        /// including gaps in the vector.
+        length: u64,
+
+        /// Max size of leaf nodes (counted in number of elements, `E`).
```

Comment:
Maybe add some guidance here in doc about how to set `max_slice_size` and `max_fan_out`? I think they affect efficiency of different operations, but I have to reach for my data structures book to remember how exactly.